src/solvers/fdsbp_tree/fdsbp_1d.jl

```diff
@@ -19,12 +19,20 @@ end
 function create_cache(mesh::TreeMesh{1}, equations,
                       volume_integral::VolumeIntegralUpwind, dg, uEltype)
 
-  prototype = Array{SVector{nvariables(equations), uEltype}, ndims(mesh)}(
-    undef, ntuple(_ -> nnodes(dg), ndims(mesh))...)
-  f_minus_threaded = [similar(prototype) for _ in 1:Threads.nthreads()]
-  f_plus_threaded = [similar(prototype) for _ in 1:Threads.nthreads()]
+  u_node = SVector{nvariables(equations), uEltype}(ntuple(_ -> zero(uEltype), Val{nvariables(equations)}()))
+  f = StructArray([(u_node, u_node)])
+  f_minus_plus_threaded = [similar(f, ntuple(_ -> nnodes(dg), ndims(mesh))...) for _ in 1:Threads.nthreads()]
+
+  f_minus, f_plus = StructArrays.components(f_minus_plus_threaded[1])
+  f_minus_threaded = [f_minus]
+  f_plus_threaded = [f_plus]
+  for i in 2:Threads.nthreads()
+    f_minus, f_plus = StructArrays.components(f_minus_plus_threaded[i])
+    push!(f_minus_threaded, f_minus)
+    push!(f_plus_threaded, f_plus)
+  end
```

Comment:
So if I understand the new cache structure `f_minus_plus_threaded` contains everything and then `f_minus_threaded` and `f_plus_threaded` are acting like "pointers" to the data actually stored in `f_minus_plus_threaded`. Is that the right way to think about this?

Reply:
Yes, that works as a mental model.